### DISCUSSION TITLE
git segment missing

### DISCUSSION THREAD

**Discussion by ffes** [JanDeDobbeleer/oh-my-posh#2726] [Troubleshoot] [answered]
Today I upgraded to the 8.36.4 but I'm not sure what the previous version was. Must have been a version in 8.x somewhere.

Since that my git segment is not showing anymore, all the other segments seem to work. The config hasn't changed since early April. I am not sure if I have missed something in the docs and need to update my config, or if this is a problem and need to open an issue.

Seeing this problem with oh-my-posh on WSL2 on Win10 in Windows Terminal (although it seems strange that it is related to the OS).

**Comment by lewis-yeung** [comment, marked as answer]
Found a [bug](https://github.com/JanDeDobbeleer/oh-my-posh/blob/0b77995d80728a75002e3501d23bf0d1b6e142ce/src/environment/shell_unix.go#L100-L106) there. I'll create a PR to fix it.

**Reply by lewis-yeung** [reply]
@ffes Please wait for #2727. After it lands, update Oh My Posh and try again. :)

**Reply by ffes** [reply]
Thanks. I see it is OS related and it was a recent change.

Obviously I will let you know

**Comment by lewis-yeung** [comment]
@ffes I guess the last working release is [v8.36.1](https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v8.36.1), right?

**Comment by ffes** [comment]
Just tested the whole v8.36.x series. With v8.36.2 it still worked, with v8.36.3 the segment disappeared.

![image](https://user-images.githubusercontent.com/3225656/187398385-d6dacddd-d9d4-40d5-8ba4-9ee18ed64843.png)

Note that I update oh-my-posh in another tab

**Reply by lewis-yeung** [reply]
@ffes Weird. How did the entire `git` segment disappear? On my end only the Git status is gone in both v8.36.2 and v8.36.3, and this only happened in a WSL shared path (e.g., `/mnt/c/path/to/git-repo`). Could you please run `oh-my-posh debug` **with v8.36.3 in your Git repo** and paste the output here?

**Reply by ffes** [reply]
[debug.txt](https://github.com/JanDeDobbeleer/oh-my-posh/files/9451818/debug.txt)

Here it is from the same directory as above

**Reply by ffes** [reply]
Just in case I attached the output of v8.36.2 as well

[oh-my-posh-debug-8.36.2.txt](https://github.com/JanDeDobbeleer/oh-my-posh/files/9451838/oh-my-posh-debug-8.36.2.txt)

**Reply by lewis-yeung** [reply]
@ffes Ah, I see the problem: 

```
2022/08/30 11:54:22 debug: RunCommand
\\wsl$\Ubuntu\home\frank\src\fesevur-nl
2022/08/30 11:54:22 RunCommand duration: 851.252µs, args: wslpath -w /home/frank/src/fesevur-nl
2022/08/30 11:54:22 GOOS duration: 49ns, args: 
2022/08/30 11:54:23 error: CommandPath
not relevant
2022/08/30 11:54:23 CommandPath duration: 12.765523ms, args: git.exe
2022/08/30 11:54:23 HasCommand duration: 12.768391ms, args: git.exe
```

The old logic didn't handle the case that the prefix of the path converted by `wslpath -w` can be `\\wsl$`, which is `\\wsl.localhost` on my end.

https://github.com/JanDeDobbeleer/oh-my-posh/blob/0b77995d80728a75002e3501d23bf0d1b6e142ce/src/environment/shell_unix.go#L105

I have updated the fix. :)

**Reply by ffes** [reply]
Could the difference between `\\wsl$` and `\\wsl.localhost` be a Win10 vs Win11 difference?

**Reply by lewis-yeung** [reply]
Maybe, I guess. I'm on Win11 and both work.

**Reply by lewis-yeung** [reply]
@ffes A [new version](https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v8.36.5) is live now.

**Reply by ffes** [reply]
On my laptop at work, where I encountered and reported the problem, v8.36.5 works as expected for repos in my `$HOME` directory. But for a git repo in `/mnt/c/Source` it doesn't work. The git segment disappears there as well. I must say I didn't try that path before, because I didn't have the time to do so yet. Hopefully tonight I will have some time to dig in this a bit more.

**Reply by lewis-yeung** [reply]
> But for a git repo in `/mnt/c/Source` it doesn't work. The git segment disappears there as well.

@ffes Actually, the answer is simple: `/mnt/c/Source` is a WSL shared path, so Oh My Posh will call the Windows Git executable (`git.exe`) instead of the Linux one (`git`). This is for performance reasons, see #1486 for details.

Do you have `appendWindowsPath = false` in the `[interop]` section of your `/etc/wsl.conf`? Or you may have disabled execute permissions of all files in Windows filesystem (e.g., `options = "metadata,umask=22,fmask=111"` in the `[automount]` section, which is what I use). Since `git.exe` is not found in your WSL PATH, which is reported in the debug logs you provided:

```
2022/08/30 11:54:23 error: CommandPath
not relevant
2022/08/30 11:54:23 CommandPath duration: 12.765523ms, args: git.exe
2022/08/30 11:54:23 HasCommand duration: 12.768391ms, args: git.exe
```

the segment will definitely not be rendered. 😆

**Reply by ffes** [reply]
Make sense. Git for Windows was installed together with Visual Studio when I started using this machine. Apparently it wasn't added to the path than. I never use GfW from anything else then the Git Bash, so I never missed it in my path either. Thanks for the info.

**Reply by ffes** [reply]
Thanks again for the quick replies and fixing the problem
